### PR TITLE
refactor!: use `this` types in service-object

### PR DESCRIPTION
**Pull request by JustinBeckwith**
A hat tip towards #426.  Soooooo not ready yet.

### CODE REVIEW

File: src/service-object.ts [hunk 1]
```diff
@@ -525,7 +520,9 @@ class ServiceObject<T = any> extends EventEmitter {
    * @param {string} reqOpts.uri - A URI relative to the baseUrl.
    * @param {function} callback - The callback function passed to `request`.
    */
-  request(reqOpts: DecorateRequestOptions): Promise<RequestResponse>;
+  request<T = ResponseBody>(
+    reqOpts: DecorateRequestOptions
+  ): Promise<RequestResponse>;
```
Comment (callmehiphop) [inline src/service-object.ts]: Should this be `Promise<RequestResponse<T>>`?

Comment (jkwlui) [reply src/service-object.ts]: I think this is half-finished - I'm going to revert this, and open another PR to genericize the request methods.

File: src/service-object.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export interface Methods {
 }
 
 export interface InstanceResponseCallback<T> {
-  (err: ApiError | null, instance?: T | null, apiResponse?: r.Response): void;
+  (err: ApiError | null, instance?: T | null, apiResponse?: Metadata): void;
```
Comment (jkwlui) [inline src/service-object.ts]: We need to decide what the correct return type for `get` is.

File: src/service-object.ts (hunk 1 above)
Comment (jkwlui) [reply src/service-object.ts]: ..as part 3 of #426

**Comment by bcoe** [comment]
@callmehiphop @JustinBeckwith what are the odds of dusting this off at some point? wonder who would be best to hand it off to ... @callmehiphop you seem to have a bit more context than me.

**Comment by JustinBeckwith** [comment]
The work that really needs to happen here is vetting the change with a few dependents, like nodejs-storage, nodejs-dns, etc.  I seem to remember not being about to make the types work when I tried it.

**Comment by callmehiphop** [comment]
@JustinBeckwith @bcoe any other clients we'd like to test against besides storage and dns? I'm happy to take a crack at this

**Comment by JustinBeckwith** [comment]
`nodejs-storage` and `cloud-profiler-nodejs` are great candidates.  They are the ones I always seem to break 😆 

In all honesty - I seem to remember coming to the conclusion this approach wouldn't work after trying to update nodejs-storage. Please feel free to correct me though!

**Comment by JustinBeckwith** [comment]
I just couldn't make this work, and this PR ain't aging like a fine wine.  Closing this out for now.

**Pull request closed without merging**